webapp/src/components/Analysis/PerturbationTestingPreview.test.tsx

```diff
@@ -0,0 +1,254 @@
+import { fireEvent, screen, waitFor } from "@testing-library/react";
+import { renderWithRouterAndRedux } from "mocks/utils";
+import PerturbationTestingPreview from "./PerturbationTestingPreview";
+import { AvailableDatasetSplits } from "types/api";
+import { setupServer } from "msw/node";
+import {
+  getPertubationResponse,
+  getPertubationResponseWithoutTrainFailureRate,
+  getPertubationResponseWithFailureResponse,
+} from "mocks/api/mockPertubationAPI";
+import { formatRatioAsPercentageString } from "utils/format";
+
+const renderPerturbationTestingPreview = (
+  availableDatasetSplits: AvailableDatasetSplits
+) =>
+  renderWithRouterAndRedux(
+    <PerturbationTestingPreview
+      jobId="local"
+      pipeline={{ pipelineIndex: 0 }}
+      availableDatasetSplits={availableDatasetSplits}
+    />,
+    { route: "/local?pipeline_index=0", path: "/:jobId?:pipeline" }
+  );
+
+describe("PerturbationTestingPreview", () => {
+  const handlers = [getPertubationResponse];
+  const server = setupServer(...handlers);
+
+  beforeAll(() => server.listen());
+  afterEach(() => server.resetHandlers());
+  afterAll(() => server.close());
+
+  it("should be having two toggle buttons for the train and test", async () => {
```

Comment:
That's a detail, but I still see two tests with "should be having" instead of "should have".
```suggestion
  it("should have two toggle buttons for the train and test", async () => {
```